### DISCUSSION TITLE
newrelic_synthetics_alert_condition deprecation - why, how to achieve same result with nrql?

### DISCUSSION THREAD

**Discussion by kirsie** [newrelic/terraform-provider-newrelic#2224] [Q&A] [answered]
Hi,

On documentation page I see that this resource will be deprecated. I think that it is really easy to use when it comes to alerting when synthetics monitoring check is failing. The recommendation -> nrql check - is much more difficult to use in this case (I am new to New Relic). Can you tell me why this is deprecated and maybe some idea how can I achieve same result using nqrl check (automatic resolver, fast alerting)

**Comment by kidk** [comment]
Hi @kirsie I've reached out internally for some examples.

**Comment by ryanv94** [comment, marked as answer]
Hi @kirsie - with NRQL alerts, especially for Synthetics - it's complicated. 

The problem is the cadence at which synthetics runs. If you have checks running minutely, that's fine. But the moment you switch to less frequent runs, then alerts NRQL gets hugely more difficult.

Therefore all examples will end up being complicated.

To the point where you need to calculate every element of a NRQL condition. Window durations need to match the frequency of the synthetic check. Which means there will likely be an extended delay between a failure occurring and an alert opening up.
Just by nature of the fact that the data is always choppy. With checks running at a cadence, it's never a smooth simple line of data here.

But, an example that you could use is, for an alert that triggers when a monitor fails, and the monitors frequency is set to 15 mins:
* **Window duration:** 15 mins.
* **Threshold**: Query returns a value above 0 at least once in 15 mins
* **Query:** `SELECT filter(count(*), WHERE result != 'SUCCESS') FROM SyntheticCheck WHERE monitorName = 'myMonitorName'`


So, something like the attached screenshot should work. But - this should be set up with the caveat that as with any data signal that's not an easy constant stream of data, the alerts will likely require some iteration to get perfectly right.

<img width="846" alt="image" src="https://user-images.githubusercontent.com/37779995/215477075-f46dd87d-1aff-4763-ae97-81af9afd593f.png">

**Reply by kirsie** [reply]
Hi,

Thanks for fast response and answering my question. Since it's pretty complicated to achieve exactly same result using NRQL what about deprecation of newrelic_synthetics_alert_condition resource? Is it something that will be deprecated only in terraform, but we will be able to use that in portal GUI? Or new relic will remove synthetic check condition completely which will result in deletion of previously created condition checks? 

I am wondering about consequences of using this resource in the future

**Reply by ryanv94** [reply]
I don't know details of the deprecation for this in the Terraform provider. But the plan long term is to move more alerts towards NRQL, while it can be a bit more complicated, it's also far more flexible. So I imagine the deprecation is around the shift to NRQL alerts, but there is no EOL timeline for the Synthetics alert condition types right now. Once we have that timeline, it'll be published. I'll leave it to @kidk to discuss the Terraform resource deprecation. 😁

**Reply by kidk** [reply]
Indeed @ryanv94 

If and when the deprecation happens @kirsie we will notify you at least 90 days before it happens, in addition I think the alerting team is planning migration steps for existing alerts. 

When we are notified ahead of time that the EOL will happen in the future, and the date is set, we will deprecate the resources in Terraform. As long as we don't have a fixed date we keep it to the warning message.

I'm not sure what the plans are for the UI, everything above is all in relation to the API.
